src/options.c

```diff
@@ -0,0 +1,363 @@
+/* Copyright (c) 2013 Jonathan Klee
+
+This file is part of ngp.
+
+ngp is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+ngp is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with ngp.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <string.h>
+#include <getopt.h>
+#include <sys/types.h>
+#include <dirent.h>
+
+#include "list.h"
+#include "utils.h"
+#include "options.h"
+
+#define NGP_VERSION   "1.4"
+
+static void usage(int status)
+{
+    FILE *out = status == 0 ? stdout : stderr;
+    fprintf(out, "usage: ngp [-h|--help] [-v|--version]\n");
+    fprintf(out, "       ngp [--<parser>] [<parser-options>] -- <pattern> [<path>]\n");
+    fprintf(out, "       ngp [--<parser>=<parser-options>] <pattern> [<path>]\n");
+    fprintf(out, "\n");
+    fprintf(out, "options:\n");
+    fprintf(out, " -h, --help     display this message\n");
+    fprintf(out, " -v, --version  show ngp version\n");
+    fprintf(out, "\n");
+    fprintf(out, "parser:\n");
+    fprintf(out, " --int[=<int-options>]       use ngp's internal search implementation with <int-options>\n");
```

Comment:
I would rather speak in terms of "native" parser.

Reply:
I, agree. Done!